src/main/resources/application.yaml

```diff
@@ -0,0 +1,14 @@
+spring:
+  sql:
+    init:
+      mode: always
+  datasource:
+    driver-class-name: org.postgresql.Driver
+    url: jdbc:postgresql://localhost:5434/${DB_NAME}
```

Comment:
That's a good point to have the database name, username and password in environment variables. But this makes me impossible to run on my localhost.
This will be a problem, as all the team members must have those environment variables set to run the application.
Instead, use hardcoded values for local run. You can have other application.yml files for production and test environments with environment variables.
Another point is that the unit tests will use this same file. If you run your unit tests agains the same database, you may encounter some problems with unit tests which change data on the database. Use a different application.yml for your unit tests, with a different database name, username and password to make sure you don't impact the local database.
Now that you have the docker-compose file, you can even create both databases, the one for local running and the one for the unit tests.